File: E2ETestResults/CoreSubset/Core/Utility/FeatureFlags.cs
// ---------------------------------------------------------------------------------------------------------------------
// <copyright file="FeatureFlags.cs" company="Tableau Software">
//   This file is the copyrighted property of Tableau Software and is protected by registered patents and other
//   applicable U.S. and international laws and regulations.
//
//   Unlicensed use of the contents of this file is prohibited. Please refer to the NOTICES.txt file for further details.
// </copyright>
// ---------------------------------------------------------------------------------------------------------------------

namespace Tableau.JavaScript.Vql.Core
{
    using System.Collections.Generic;
    using System.Diagnostics.CodeAnalysis;
    using Tableau.JavaScript.Vql.TypeDefs;

    public static class FeatureFlags
    {
        private static JsDictionary<FeatureFlagIds, bool> defaultValues;

        public static bool IsEnabled(FeatureFlagIds featureFlagId)
        {
            if (TsConfig.Features != null && TsConfig.Features.ContainsKey(featureFlagId))
            {
                return TsConfig.Features[featureFlagId];
            }

            if (defaultValues != null && defaultValues.ContainsKey(featureFlagId))
            {
                return defaultValues[featureFlagId];
            }

            return false;
        }

        // used in tests (see FeatureFlagDefaults.cs)
        public static void SetDefaults(JsDictionary<FeatureFlagIds, bool> defaults)
        {
            defaultValues = defaults;
        }
    }
}

File: E2ETestResults/CoreSubset/Core/Utility/FeatureFlags.ts
import { FeatureFlagIds, tsConfig } from 'TypeDefs';

export class FeatureFlags {
  private static defaultValues: { [key: string]: boolean };

  public static isEnabled(featureFlagId: FeatureFlagIds): boolean {
    if (tsConfig.features !== null && ss.keyExists(tsConfig.features, featureFlagId)) {
      return tsConfig.features[featureFlagId];
    }
    if (FeatureFlags.defaultValues !== null && ss.keyExists(FeatureFlags.defaultValues, featureFlagId)) {
      return FeatureFlags.defaultValues[featureFlagId];
    }
    return false;
  }

  public static setDefaults(defaults: { [key: string]: boolean }): void {
    FeatureFlags.defaultValues = defaults;
  }
}
